clash-lib/src/Clash/Normalize/Strategy.hs

```diff
@@ -131,7 +131,7 @@ constantPropagation =
     inlineNR :: NormRewrite
     inlineNR =
           bottomupR (apply "deadCode" deadCode)
-      >-! bottomupR (apply "inlineNonRep" inlineNonRep)
```

Comment:
Why was simply
```haskell
>-! topdownSucR (apply "inlineNonRep" inlineNonRep)
```
not an option?

Reply:
Ah, I see. Because you want to stop the traversal in the alternatives.

Reply:
I don't know, it depends. My understanding is that `topdownSucR` stops at the first success, meaning that it will either replace zero or one non-representable case subjects and never more than one. This can get the transformation stuck in an infinite loop, for example in:

```haskell
let 
  x = case nonRep of ...
  y = case nonRep of ...
in
  case y of
    ...
```

..it will keep expanding `x` (if it's recursive) even though it might need `y` to make progress instead.

Reply:
Just to clarify, if it doesn't that'd also be subtly different from what the current transformation does. The current transformation doesn't traverse if _anything_ changes in a subject of a case, because it might be enough to make progress in `inlineAndPropagate`.